Assignment 2_Joellyn Heng.Rmd

```diff
@@ -77,24 +102,116 @@ plot(g,layout=layout.fruchterman.reingold)
 
-````
+```
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+stud_count_df <- EDGE %>%
+  dplyr::group_by(to) %>%
+  dplyr::summarise(sum(count)) %>%
+  as.matrix()
+
+teacher_count_df <- EDGE %>%
+  dplyr::group_by(from) %>%
+  dplyr::summarise(sum(count)) %>%
+  as.matrix()
+
+combined_count <- rbind(stud_count_df, teacher_count_df)
+
+combined_count <- as.data.frame(combined_count)
+combined_count$`sum(count)` <- as.numeric(combined_count$`sum(count)`)
 
+V(g)$size <- combined_count$`sum(count)`*2
+
+plot(g, vertex.color=VERTEX$gender)
+
+```
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
+```{r set up, include = FALSE}
+setwd("~/Documents/GitHub/EDM Class/assignment2")
+DF_class <- read.csv("Data/hudk4050-classes.csv")
+library(dplyr)
+library(tidyverse)
+```
+
+```{r tidy data set}
+DF_class <- DF_class %>%
+  select("First_N" = Q8, "Last_N" = Q9, "UNI" = Q10, "Class 1" = Q1, "Class 2" = Q3, "Class 3" = Q4, "Class 4" = Q5, "Class 5" = Q6, "Class 6" = Q7) %>%
+  unite(Name, First_N, Last_N, sep = " ")
+
+DF_class <- DF_class[-c(1,2), ]
+
+for (i in 3:8){
+  DF_class[,i] <- gsub( " ", "", DF_class[,i]) 
+}
+
+DF_class[DF_class=="4050"] <- "HUDK4050"
+DF_class[DF_class=="4125"] <- "HUDM4125"
+DF_class[DF_class=="QMSS-5072"] <- "QMSS5072"
+DF_class[DF_class=="5126"] <- "HUDM5126"
+DF_class[DF_class== c("G5067", "QMSSGR5067")] <- "QMSS5067"
+DF_class[DF_class=="G5072"] <- "QMSS5072"
+DF_class[DF_class=="HUDK4050"] <- ""
+
+#rename: 4050 -> HUDK4050, 4125 -> HUDM4125, QMSS-5072 -> QMSS5072,  5126 -> HUDM5126, G5067 -> QMSS5067, G5072 > QMSS5072, QMSSGR5067 -> QMSS5067
+#left it: 5026 - not sure which dept
+#remove: HUDK4050 -> ""
+
+DF_class <- gather(DF_class, ClassNo, Classes, 3:8)
+DF_class <- DF_class[order(DF_class$Name),]
+
+DF_class <- DF_class[!(DF_class$Classes == ""), ] %>%
+  select(Name, Classes)
+
+DF_class$attend <- 1
+
+DF_class <- DF_class %>%
+  spread(Classes, attend) 
+
+DF_class[is.na(DF_class)] <- 0
+
+rownames(DF_class) <- DF_class$Name
+
+DF_class <- DF_class %>%
+  select(-Name)
+
+DF_class_matrix <- as.matrix(DF_class)
+adj_matrix <- DF_class_matrix %*% t(DF_class_matrix)
+adj_matrix
+```
+
+```{r person network convert data to igraph network object}
+network <- graph_from_adjacency_matrix(adj_matrix)
+network
+```
+
+```{r assign attributes and plot}
+E(network)$arrow.mode <- 0
+networks <- simplify(network)
+plot(networks, vertex.size = 5)
+
```

Comment:
It looks like you have not removed the diagonal in the matrix so everyone is still connected to themselves. You may want to also change the font size on the labels to better see the network underneath.